CMakeLists.txt

```diff
@@ -141,6 +141,12 @@ set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 9)
 set(NVIM_VERSION_PATCH 0)
 set(NVIM_VERSION_PRERELEASE "-dev") # for package maintainers
+# Allow manually setting git tag for situations where `git describe` cannot
+# work such as the nix flake.
```

Comment:
why _cannot_ `git describe` work in situations such as the nix flake? Asking for a maintainer.

Reply:
I'm new to nix myself so I'm not fully sure. I fond the `self.shortRev or "dirty"` pattern in the .nix file from searching how other people got git revs in their builds, so I mostly assumed it was not possible to run git commands, backed up by the calls presumably working in neovim's cmake scripts when it was in nix.

I can guess at two reasons which would make `git describe` not work if either or perhaps both are true.

1. The git command isn't being pulled into the environment in which the build is happening. If this is the case, then it might not be so bad to get it to work, "just" adding it to some list of tools to pull in. Though even that is currently beyond my nix knowledge. 

2. The .git directory doesn't make it into the build environment. I could see a situation where maybe the nix functions that pull down the branch to wherever that happens only pulls the files in the branch and not the .git directory. It seems like they'd be going out of their way to do this, but it would also explain `git describe` not working. If this is the case, it would be a lot more work to fix than if it's just #1. 

It could of course be some other reason(s) too. Sorry I can't be of more help for this question.

Reply:
@will have you tried adding git to the buildInputs ? you might get more help in nix channels than neovim's.

Reply:
I have now managed to do include git, and also change `execute_process( COMMAND git...` to show both stderr as well, and it looks like both of my guesses were true.

```
> nix run ./contrib/ -- --version
warning: Git tree '/private/tmp/neovim' is dirty
NVIM v0.9.0-dev-fatal: not a git repository (or any of the parent directories): .git
Build type: Release
```

From the nix forums, it looks like the reason the .git directory isn't there is because it fetches the source as a tarball and not through git at all which doesn't include the .git directory: https://discourse.nixos.org/t/keep-git-folder-in-when-fetching-a-git-repo/8590

and so the only way is with `self.shortRev` and getting that info into the build some other way: https://discourse.nixos.org/t/accessing-git-directory-in-flake-with-local-source/17370

Reply:
you can swap the fetchFromGithub call to fetchGit with keep https://discourse.nixos.org/t/keep-git-folder-in-when-fetching-a-git-repo/8590/3 . Alternatively you could patch the source in a postPatch/preConfigure phase to replace `set(NVIM_VERSION_PRERELEASE "-dev") # for package maintainers` this line with the git version ? 

we dont necessarily need to modify the makefile, and if we do it should not be to pass a GIT_TAG but a generic prerelease string (which can be the git tag)

Reply:
after discussion with @dundargoc , doing everything in nix seems better. Looking at the code I think patching the generated file https://github.com/neovim/neovim/blob/98daaa798e018071876d026a60840991be8d8069/cmake.config/versiondef.h.in#L7 after the configurePhase to insert self.shortRev into it is the best way

Reply:
@teto (and @dundargoc) thanks for the feedback, and especially the pointer to exactly what needed to be patched. 

I've updated the PR to just patch that one spot and it works the same and is much cleaner:

```
$ nix run ./contrib/ -- --version
NVIM v0.9.0-dev-cafe905
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by _nixbld1
```

Reply:
great ! works for me